### ISSUE TITLE
New matcher: create contact AND contribution

### ISSUE THREAD

**Issue by systopia** [Project60/org.project60.banking#123]
Pretty much what it says in the title

**Comment by systopia** [comment]
Still needs to be pulled into ``master``...

@VangelisP: Could you create a pull request?

**Comment by CS-DaS** [comment]
We would like to contribute this matcher. We already created a working prototype and currently are finishing the work by adding tests, documentation, error handling, translation, make it more configurable, ...

Would you be up to include this new matcher into the upcoming version 1.5.0? Is there a time frame until when we should finish the work to get it into 1.5.0?

Do you prefer that we first finish the whole thing and then create a pull request, or would you like to participate more closely and we should create a PR already now with the already existing prototype?

It's basically an adapted copy of the CreateContribution-matcher with some hard-coded logic to extract name parts from the `name` field.

**Comment by dontub** [comment]
Thank you for working on this issue!

> Would you be up to include this new matcher into the upcoming version 1.5.0? Is there a time frame until when we should finish the work to get it into 1.5.0?

We're generally open to add new matchers. When it is part of the master branch it'll be part of the next release - whether it is 1.5.0 or a later version. I think a new matcher would justify a new release in a timely manner. Please keep in mind that also reviewing code takes time which might sometimes be difficult to be done for free (depending on the amount of code).

> Do you prefer that we first finish the whole thing and then create a pull request, or would you like to participate more closely and we should create a PR already now with the already existing prototype?

It might make sense to get feedback before everything is finished. Feel free to make a PR when you think that the code is in an appropriate state.

> It's basically an adapted copy of the CreateContribution-matcher with some hard-coded logic to extract name parts from the `name` field.

If it makes sense, consider to avoid duplicate code through refactoring. Keep DRY :-)

**Comment by CS-DaS** [comment]
I created a PR, see https://github.com/Project60/org.project60.banking/pull/558

Feel free to already have a look at the current state :)

**Comment by VangelisP** [comment]
I'm also happy to review your matcher if that would be helpful.

We've actually been running our own version of this matcher as a separate extension for several years, and I'd say quite successfully. Unfortunately, I never got around to submitting a PR for it, as at the time I assumed we were probably the only ones who needed this functionality.

Over the last year or so, I've also refactored our version to support APIv4 directives alongside the original APIv3 ones, so I'd be happy to review your approach and suggest improvements where relevant. I can also share parts of our implementation if that's useful.

**Comment by CS-DaS** [comment]
@VangelisP Yes that would be awesome if you could review the PR! And if you have something to share which helps, that would also be great!

**Comment by VangelisP** [comment]
@CS-DaS I want to share a markdown documentation on potential suggestions/improvements however, on your forked git repo the issues are turned off. Any suggested space to share my thoughts ? I don't want to abuse civibanking's issue queue for that

**Comment by CS-DaS** [comment]
Thanks for your feedback! I activated the issues in my fork repo.
